src/model/operation/transform.js

```diff
@@ -71,47 +112,142 @@ export function transform( a, b, context = {} ) {
 	}
 }
 
+/**
+ * Performs a transformation of two sets of operations - `operationsA` and `operationsB`. The transformation is two-way -
+ * both transformed `operationsA` and transformed `operationsB` are returned.
+ *
+ * Note, that the first operation in each set should base on the same context. A simplified version of context is
+ * {@link module:engine/model/document~Document#version document version}, so the
+ * {@link module:engine/model/operation/operation~Operation#baseVersion base version} of the first operation from `operationsA`
+ * has to be the same as the base version of the first operation from `operationsB`.
+ *
+ * It is assumed that `operationsA` are "more important" during conflict resolution between two operations.
+ *
+ * New copies of both passed arrays and operations inside them are returned. Passed arguments are not altered.
+ *
+ * Base versions of the transformed operations sets are updated accordingly. For example, if base versions start from `4`
+ * and there are three operations in `operationsA` and five operations in `operationsB`, then transformed `operationsA`
+ * will start from base version `9` while transformed `operationsB` will start from base version `7`. If no operation was
+ * broken into two during transformation, then both sets will end up with an operation that bases on version `11`.
```

Comment:
I had to read this 3 times to get it. What can be improved: `three` -> `3`, `five` -> `5`. Some "math helpers":
 - `9 (4 base + 5 operations A)`,
 - `7 (4 base + 3 operations B)`,
 - `11 (4 base + 5 operations A + 3 operation B trandormed by A = 4 base + 3 operations B + 3 operation A trandormed by B = 12, what means that the base of the last operation is 11)`,